### PR TITLE
On Linux, add header feature flag to fix version regression

### LINKED ISSUE #5625: [bug] Tauri not building an app on Linux Ubuntu 18.04 due to dependency mismatch
### Describe the bug

I am using Ubuntu version 18.04 which has `libwebkit2gtk-4.0` version 2.32.4 and there's no way to securely update that. Only newer versions of Ubuntu have the required `webkit2gtk-4.0 >= 2.36`.

### Reproduction

1.Try building app on Linux Ubuntu version 18.04

### Expected behavior

I expect it to build without any errors. 

### Platform and versions

yarn tauri info                                                                                                                                                      (101) (*main+1) 13:48:09 
yarn run v1.22.19
$ tauri info

Environment
  › OS: Ubuntu 18.04 X64
  › Node.js: 16.17.0
  › npm: 8.15.0
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.65.0
  › cargo: 1.65.0
  › Rust toolchain: stable-x86_64-unknown-linux-gnu 

Packages
WARNING: no lock files found, defaulting to npm
  › @tauri-apps/cli [NPM]: 1.2.0
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.0,
  › tauri-build [RUST]: 1.2.0,
  › tao [RUST]: 0.15.3,
  › wry [RUST]: 0.22.0,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:1420/
  › framework: React (Next.js)
  › bundler: Webpack

App directory structure
  ├─ .vscode
  ├─ .next
  ├─ src
  ├─ node_modules
  └─ src-tauri
Done in 10.37s.

### Stack trace

```shell
$ yarn tauri dev                                                                                                                                                       (130) (*main+1) 13:47:57 
yarn run v1.22.19
$ tauri dev
     Running BeforeDevCommand (`yarn dev`)
$ next dev -p 1420
ready - started server on 0.0.0.0:1420, url: http://localhost:1420
        Info Watching /var/apps/desktop/src-tauri for changes...
   Compiling webkit2gtk-sys v0.18.0
   Compiling tao v0.15.3
   Compiling tauri-runtime v0.12.0
   Compiling thread_local v1.1.4
   Compiling pathdiff v0.2.1
   Compiling fastrand v1.8.0
   Compiling remove_dir_all v0.5.3
   Compiling tar v0.4.38
The following warnings were emitted during compilation:

warning: `"pkg-config" "--libs" "--cflags" "webkit2gtk-4.0" "webkit2gtk-4.0 >= 2.36"` did not exit successfully: exit status: 1

error: failed to run custom build command for `webkit2gtk-sys v0.18.0`

Caused by:
  process didn't exit successfully: `/var/apps/desktop/src-tauri/target/debug/build/webkit2gtk-sys-de24a0cc7120fb4a/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=WEBKIT2GTK_4.0_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:warning=`"pkg-config" "--libs" "--cflags" "webkit2gtk-4.0" "webkit2gtk-4.0 >= 2.36"` did not exit successfully: exit status: 1
  error: could not find system library 'webkit2gtk-4.0' required by the 'webkit2gtk-sys' crate

  --- stderr
  Requested 'webkit2gtk-4.0 >= 2.36' but version of WebKitGTK is 2.32.4
  You may find new versions of WebKitGTK at https://webkitgtk.org

warning: build failed, waiting for other jobs to finish...
event - compiled client and server successfully in 800 ms (176 modules)nd, thread_local, tar, pathdiff, tao, remove_dir_all                                                                                      

error Command failed with exit code 101.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wusyong**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information
fix https://github.com/tauri-apps/tauri/issues/5625

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -64,6 +64,8 @@
 //! Avoid this in release build if your app needs to publish to App Store.
 //! - `dox`: Enables this in `package.metadata.docs.rs` section to skip linking some **Linux**
 //! libraries and prevent from building documentation on doc.rs fails.
+//! - `header`: Enables headers support of custom protocol request on Linux. Requires webkit2gtk
+//! v2.36 above.
```
Comment (lucasfernog) [inline src/lib.rs]:
```suggestion
//! v2.36 or above.
```

File: Cargo.toml [hunk 2]
```diff
@@ -31,6 +31,7 @@ tray = [ "tao/tray" ]
 devtools = [ ]
 transparent = [ ]
 fullscreen = [ ]
+header = [ "webkit2gtk/v2_36" ]
```
Comment (lucasfernog) [inline Cargo.toml]: maybe we should call it `linux-headers` or `linux-custom-protocol-headers`?

Comment (FabianLars) [reply Cargo.toml]: or maybe just make version features similar to webkit2gtk, for example the "audio without user interaction" fix/feature needs 2.36 apis too

Comment (wusyong) [reply Cargo.toml]: @FabianLars Do you have the link for audio link as reference?

Comment (FabianLars) [reply Cargo.toml]: The test branch still exists: https://github.com/tauri-apps/wry/compare/dev...autoplay-0.19
i'll add a link to the docs to this comment once i find it again

Comment (FabianLars) [reply Cargo.toml]: Oh nevermind it was 2.30...

Comment (wusyong) [reply Cargo.toml]: I'll pretend I didn't see that. :see_no_evil: 
@lucasfernog and I want to do some workaround in webkit2gtk and we can just check version during runtime.
After that, I'll switch flatpak and bring to v2 soon.

Comment (wusyong) [reply Cargo.toml]: I tried to remove feature flags in webkit2gtk-rs but there are too many. I fear that will cause trouble in future updates. I would like to ask to consider this PR again.

Comment (lucasfernog) [reply Cargo.toml]: Let's do it this way for now. if we need more version-related feature flags, we can either change it or recommend users to add the webkit2gtk dependency and define the feature they need.

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**